architecture/high-availability.md

```diff
@@ -0,0 +1,33 @@
+# High Availability
+
+Kubecost 2.2 introduces a new flag `haMode` for the frontend service.
+
+This is the first step in enabling high availability for Kubecost. Future versions will expand the HA configuration to include the backend services.
+
+> This flag changes the service name that the ingress needs to target.
+
+With the Frontend HA Mode, the frontend will now be a dedicated deployment with two replicas. This allows for the ability to do rolling upgrades of the frontend pods during upgrades and configuration changes.
+

```

Comment:
Instead of "frontend service", can we say "product UI," or is there a difference? We rarely mention the phrase frontend service in this way through our docs.